src/catalog/catalog_entry/duck_table_entry.cpp

```diff
@@ -754,6 +726,51 @@ unique_ptr<CatalogEntry> DuckTableEntry::DropForeignKeyConstraint(ClientContext
 	return make_uniq<DuckTableEntry>(catalog, schema, *bound_create_info, storage);
 }
 
+unique_ptr<CatalogEntry> DuckTableEntry::AddConstraint(ClientContext &context, AddConstraintInfo &info) {
+	auto create_info = make_uniq<CreateTableInfo>(schema, name);
+	create_info->comment = comment;
+
+	// Copy all columns and constraint to the modified table
+	create_info->columns = columns.Copy();
+	for (auto &constraint : constraints) {
+		create_info->constraints.push_back(constraint->Copy());
+	}
+
+	switch (info.constraint->type) {
+	case ConstraintType::UNIQUE: {
+		auto unique = reinterpret_cast<UniqueConstraint *>(info.constraint.get());
+
+		if (unique->is_primary_key) {
+			EnsureNoPrimaryKey();
+		}
+
+		create_info->constraints.push_back(info.constraint->Copy());
+		break;
+	}
+	default:
+		throw InternalException("Unsupported constraint type in ALTER TABLE statement!");
+	}
+
+	auto binder = Binder::CreateBinder(context);
+	// We need to create a new physical table with a new constraint and possibly a new
+	// unique index. This procedure is also responsible for checking that the existing
+	// data satisfies the new constraint.
+	auto bound_constraint = binder->BindConstraint(*info.constraint.get(), create_info->table, create_info->columns);
```

Comment:
`*info.constraint`

Using `.get()` gives a raw pointer and evades the null check, causing this to segfault instead of throwing an InternalException if `constraint` were to be null

Reply:
I also replaced all occurrences of this with de-referencing.